src/search_ctx.h

```diff
@@ -23,43 +23,44 @@ extern "C" {
 #define CLOCK_MONOTONIC_RAW CLOCK_MONOTONIC
 #endif
 
+typedef enum {
+  RS_CTX_UNSET,
+  RS_CTX_READONLY,
+  RS_CTX_READWRITE
+} RSContextFlags;
+
+
 /** Context passed to all redis related search handling functions. */
 typedef struct RedisSearchCtx {
   RedisModuleCtx *redisCtx;
   RedisModuleKey *key_;
   IndexSpec *spec;
-  uint32_t refcount;
-  int isStatic;
   uint64_t specId;  // Unique id of the spec; used when refreshing
   struct timespec timeout;
   unsigned int apiVersion; // API Version to allow for backward compatibility / alternative functionality
+  RSContextFlags flags;
+  bool isLocked;
```

Comment:
Is `isLocked` redundant?
(`flags` == `RS_CTX_UNSET` actually means also `isLocked` == 0 ?)

Reply:
yes